src/engine.c

```diff
@@ -881,6 +881,24 @@ ENGINE_getKeyState(ENGINE* engine, char* keyName) {
   return state[scancode];
 }
 
+internal uint32_t
+ENGINE_findMouseCursorIndex(ENGINE* engine, char* cursorName) {
+  int cursorIndex;
+  for (int index = 0; index < 12; index++) {
+    char * name = ENGINE_MOUSE_CURSORS[index];
+    if (strcmp(name, cursorName) == 0) cursorIndex = index;
+  }
+  return cursorIndex;
+}
+
+internal void
+ENGINE_setMouseCursor(ENGINE* engine, char* cursorName) {
+  SDL_FreeCursor(engine->mouse.cursor);
+  int cursorIndex = ENGINE_findMouseCursorIndex(engine, cursorName);
+  engine->mouse.cursor = SDL_CreateSystemCursor(cursorIndex);
```

Comment:
We need to `SDL_FreeCursor` the previous cursor before doing this. (To do that safely, we should also initialise this property to NULL). We also need to do the free during `ENGINE_free`.

Reply:
It'd be nice I think if we could cache this value, but we don't have the infrastructure for it at the moment.

Reply:
So when I'm calling `SDL_FreeCursor(engine->mouse.cursor);` you're saying it's not actually doing anything? Or should i be calling that somewhere else?

Reply:
Oh..no, I'm just blind, turns out. Sorry, I didn't see the free there first time around. We need to call it in ENGINE_free though.

Reply:
No worries! Fixed in 8613623